frontend/src/components/GemMap.vue

```diff
@@ -0,0 +1,81 @@
+<template>
+  <div>
+    <h4>Hello Friendos</h4>
+    Latitude: {{ currPosition.lat.toFixed(2) }}, Longitude:
+    {{ currPosition.lng.toFixed(2) }}
+  </div>
+
+  <google-map-loader
+    :mapConfig="mapConfig"
+    :apiKey="apiKey"
+    :markersOptions="markersOptions"
+  />
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, PropType } from "vue";
+import { useGeolocation } from "../useGeolocation";
+import GoogleMapLoader from "./GoogleMapLoader.vue";
+
+const DEFAULT_MAP_CONFIG = {
+  center: { lat: 1.3521, lng: 103.8198 },
+  zoom: 13,
+};
+
+const GOOGLE_API_KEY = process.env.VUE_APP_GOOGLE_API_KEY;
+
+// TODO: replaced this once Gem interface has been created
+interface TempGem {
+  position: { lat: number; lng: number };
+}
+
+export default defineComponent({
+  props: {
+    gems: {
+      type: Array as PropType<TempGem[]>,
+      // TODO: replace this with empty array once gems can be fetched
+      // from an actual data source
+      default: () => [
+        { position: { lat: 1.2966, lng: 103.7764 } },
+        { position: { lat: 1.3483, lng: 103.6831 } },
+        { position: { lat: 1.3644, lng: 103.9915 } },
+        { position: { lat: 1.4382, lng: 103.7891 } },
+        { position: { lat: 1.3109, lng: 103.7952 } },
+        { position: { lat: 1.3309, lng: 103.8752 } },
+      ],
+    },
+  },
+  components: {
+    GoogleMapLoader,
+  },
+
+  setup() {
+    const { coords } = useGeolocation();
```

Comment:
open qn: behaviour if user clicks 'Deny' on geolocation permission? we kick them out? hahaha but out of scope probably

Reply:
yep, sounds good! I've replaced it with a simple `Loading...` message for now, but we can definitely make it more helpful (eg. how to re-enable permission if they accidentally pressed deny). I've added a TODO for this